test/Microsoft.ML.Core.Tests/UnitTests/TestVBuffer.cs

```diff
@@ -21,5 +31,1174 @@ public void TestApplyAt()
+                            dropSlotDenseOpCount++;
+                        else
+                            dropSlotSparseOpCount++;
+
+                        var expectedVector = new VBuffer<float>(Math.Max(1, expectedLength),
+                            expectedIndices.Count, expectedValues.ToArray(), a.IsDense ? null : expectedIndices.ToArray());
+
+                        var dst = rgen.Next(2) == 0 ? a : default(VBuffer<float>);
+                        slotDropper.DropSlots(ref a, ref dst);
+                        TestSame(ref expectedVector, ref dst);
+                        minSlots.Clear();
+                        maxSlots.Clear();
+                        max = -1;
+                        len = dst.Length;
+                        Utils.Swap(ref a, ref dst);
+                    }
+                }
+            }
+        }
+
+        private static float ScaleFactor(int trial, Random rgen)
+        {
+            switch (trial % 4)
+            {
+            case 0:
+                return 0;
+            case 1:
+                return 1;
+            case 2:
+                return -1;
+            default:
+                return rgen.NextDouble().ToFloat() * 10 - 5;
+            }
+        }
+
+        private static void GenerateSingle(Random rgen, int len, out VBuffer<float> a)
+        {
+            int count = rgen.Next(2) == 0 ? len : rgen.Next(len);
+            GenerateVBuffer(rgen, len, count, out a);
+        }
+
+        private static void GenerateVBuffer(Random rgen, int len, int count, out VBuffer<float> dst)
+        {
+            const int excess = 4;
+            Contracts.Assert(count <= len);
+            int[] indices = null;
+            if (count != len)
+            {
+                indices = new int[count + rgen.Next(excess)];
+                FillRandomIndices(rgen, indices, len, count);
+                for (int i = count; i < indices.Length; ++i)
+                    indices[i] = rgen.Next(200) - 100;
+                if (indices.Length == 0 && rgen.Next(2) == 0)
+                    indices = null;
+            }
+            float[] values = new float[count + rgen.Next(excess)];
+            for (int i = 0; i < values.Length; ++i)
+                values[i] = rgen.NextDouble().ToFloat() * 10 - 5;
+            if (values.Length == 0 && rgen.Next(2) == 0)
+                values = null;
+            dst = new VBuffer<float>(len, count, values, indices);
+        }
+
+        private static void FillRandomIndices(Random rgen, int[] indices, int len, int count)
+        {
+            Contracts.Assert(Utils.Size(indices) >= count);
+            int max = len - count + 1;
+            for (int i = 0; i < count; ++i)
+                indices[i] = rgen.Next(max);
+            Array.Sort(indices, 0, count);
+            for (int i = 0; i < count; ++i)
+                indices[i] += i;
+            Contracts.Assert(Utils.IsIncreasing(0, indices, count, len));
```

Comment:
Maybe we should use `Assert` here - that way the tests always fail (even in Release mode).